lib/page-object/element.js

```diff
@@ -1,24 +1,169 @@
+  var lookupList = getAncestorsWithElement(this);
+
+  if (lookupList.length > 1) {
+    return new Element({
+      selector: lookupList,
+      locateStrategy: 'recursion'
+    });
+  }
+
+  return null;
+};
+
+/**
+ * Copies selector properties to the first object from the second if the first
+ * object has undefined values for any of those properties.
+ *
+ * @param {Object} target The object to assign values to.
+ * @param {Object} source The object to capture values from.
+ */
+Element.copyDefaults = function(target, source) {
+  var props = ['name', 'parent', 'selector', 'locateStrategy'];
+  props.forEach(function(prop) {
+    if (target[prop] === undefined || target[prop] === null) {
+      target[prop] = source[prop];
+    }
+  });
+};
+
+/**
+ * Parses the value/selector parameter of an element command creating a
+ * new Element instance with the values it contains, if any. The standard
+ * format for this is a selector string, but additional, complex formats
+ * in the form of an array or object are also supported.
+ *
+ * @param {string|Object} value Selector value to parse into an Element.
+ * @param {string} [using] The using/locateStrategy to use if the selector
+ *    doesn't provide one of its own.
+ */
+Element.fromSelector = function(value, using) {
+
+  if (!value) {
+    throw new Error('Invalid selector value specified');
+  }
+
+  if (value instanceof Element) {
+    value.locateStrategy = value.locateStrategy || using;
+    return value;
+  }
+
+  if (using === 'recursion') {
+    return new Element({
+      selector: value,
+      locateStrategy: 'recursion'
+    });
+  }
+
+  var definition = {};
+
+  if (value && typeof value === 'object') {
+
+    definition.selector = value.selector;
+    if ('locateStrategy' in value) {
+      definition.locateStrategy = value.locateStrategy;
+    }
+
+  } else {
+    definition.selector = value;
+  }
+
+  definition.locateStrategy = definition.locateStrategy || using;
+
+  return new Element(definition);
 };
 
-module.exports = Element;
\ No newline at end of file
+/**
+ * Gets the name of an element instance using "(anonymous)"
+ * if a name was not defined - the case when the element was
+ * created anonymously for a command rather than being a
+ * named element in a page or section definition.
+ */
+function getName(instance) {
+  return instance.name || '(anonymous)';
+}
+
+/**
+ * Array filter method removing elements that may be defined (in) but
+ * do not have a recognizable value.
+ */
+function hasValidValueIn (definition) {
+  return function(key) {
+    return definition[key] !== undefined && definition[key] !== null;
+  };
+}
+
+/**
+ * Retrieves an array of ancestors of the supplied element. The last element in the array is the element object itself
+ *
+ * @param {Object} element The element
+ * @returns {Array}
+ */
+function getAncestorsWithElement(element) {
+  var elements = [];
+  function addElement(e) {
```

Comment:
@beatfactor which part here? addElement? getAncestorsWithElement?  This was taken directly from `command-wrapper` but was specific to element behavior so I moved it here.

https://github.com/nightwatchjs/nightwatch/blob/v0.9.8/lib/page-object/command-wrapper.js#L172